equinix_metal/equinix_metal/models/vlan_virtual_circuit.py

```diff
@@ -37,69 +39,84 @@ class VlanVirtualCircuit(BaseModel):
+    tags: Optional[List[StrictStr]] = None
     type: Optional[StrictStr] = None
     updated_at: Optional[datetime] = None
     virtual_network: Optional[Href] = None
     vnid: Optional[StrictInt] = None
-    __properties = ["bill", "bill_type", "created_at", "description", "href", "id", "name", "nni_vlan", "port", "project", "speed", "status", "tags", "type", "updated_at", "virtual_network", "vnid"]
+    __properties: ClassVar[List[str]] = ["bill", "bill_type", "created_at", "description", "href", "id", "name", "nni_vlan", "port", "project", "speed", "status", "tags", "type", "updated_at", "virtual_network", "vnid"]
 
-    @validator('bill_type')
+    @field_validator('bill_type')
     def bill_type_validate_enum(cls, value):
         """Validates the enum"""
         if value is None:
             return value
 
-        if value not in ('metal_billed', 'fabric_billed'):
+        if value not in set(['metal_billed', 'fabric_billed']):
             raise ValueError("must be one of enum values ('metal_billed', 'fabric_billed')")
         return value
 
-    @validator('status')
+    @field_validator('status')
     def status_validate_enum(cls, value):
         """Validates the enum"""
         if value is None:
             return value
 
-        if value not in ('pending', 'waiting_on_customer_vlan', 'activating', 'changing_vlan', 'deactivating', 'deleting', 'active', 'expired', 'activation_failed', 'changing_vlan_failed', 'deactivation_failed', 'delete_failed'):
+        if value not in set(['pending', 'waiting_on_customer_vlan', 'activating', 'changing_vlan', 'deactivating', 'deleting', 'active', 'expired', 'activation_failed', 'changing_vlan_failed', 'deactivation_failed', 'delete_failed']):
             raise ValueError("must be one of enum values ('pending', 'waiting_on_customer_vlan', 'activating', 'changing_vlan', 'deactivating', 'deleting', 'active', 'expired', 'activation_failed', 'changing_vlan_failed', 'deactivation_failed', 'delete_failed')")
         return value
 
-    @validator('type')
+    @field_validator('type')
     def type_validate_enum(cls, value):
         """Validates the enum"""
         if value is None:
             return value
 
-        if value not in ('vlan', 'vrf'):
```

Comment:
This is an example of the benefits of upgrading: previously, the `type` field could not be used to determine that a `VrfVirtualCircuit` _is not_ a `VlanVirtualCircuit` because the `VlanVirtualCircuit` model accepted `vrf` as a valid value for `type`.